etherman/etherman.go

```diff
@@ -648,8 +656,12 @@ func (etherMan *Client) GetTxReceipt(ctx context.Context, txHash common.Hash) (*
 func (etherMan *Client) ApproveMatic(maticAmount *big.Int, to common.Address) (*types.Transaction, error) {
 	tx, err := etherMan.Matic.Approve(etherMan.auth, etherMan.SCAddresses[0], maticAmount)
 	if err != nil {
+		if parsedErr, ok := tryParseContractPoEError(err); ok {
```

Comment:
You are trying to parse erc20 error with Poe function. Maybe you should rename the function with a more generic name. Same comment in other functions
```suggestion
		if parsedErr, ok := tryParseError(err); ok {
```

Reply:
I've updated it please review it again. :)